server/src/test/java/org/elasticsearch/env/OverwriteNodeVersionCommandTests.java

```diff
@@ -0,0 +1,155 @@
+            new OverwriteNodeVersionCommand().processNodePaths(mockTerminal, new Path[]{emptyPath}, environment));
+        assertThat(elasticsearchException.getMessage(), equalTo(OverwriteNodeVersionCommand.NO_METADATA_MESSAGE));
+        expectThrows(IllegalStateException.class, () -> mockTerminal.readText(""));
+    }
+
+    public void testFailsIfUnnecessary() throws WriteStateException {
+        final Version nodeVersion = Version.fromId(between(Version.CURRENT.minimumIndexCompatibilityVersion().id, Version.CURRENT.id));
+        NodeMetaData.FORMAT.writeAndCleanup(new NodeMetaData(randomAlphaOfLength(10), nodeVersion), nodePaths);
+        final MockTerminal mockTerminal = new MockTerminal();
+        final ElasticsearchException elasticsearchException = expectThrows(ElasticsearchException.class, () ->
+            new OverwriteNodeVersionCommand().processNodePaths(mockTerminal, nodePaths, environment));
+        assertThat(elasticsearchException.getMessage(), allOf(
+            containsString("compatible with current version"),
+            containsString(Version.CURRENT.toString()),
+            containsString(nodeVersion.toString())));
+        expectThrows(IllegalStateException.class, () -> mockTerminal.readText(""));
+    }
+
+    public void testWarnsIfTooOld() throws Exception {
+        final String nodeId = randomAlphaOfLength(10);
+        final Version nodeVersion = NodeMetaDataTests.tooOldVersion();
+        NodeMetaData.FORMAT.writeAndCleanup(new NodeMetaData(nodeId, nodeVersion), nodePaths);
+        final MockTerminal mockTerminal = new MockTerminal();
+        mockTerminal.addTextInput("n\n");
+        final ElasticsearchException elasticsearchException = expectThrows(ElasticsearchException.class, () ->
+            new OverwriteNodeVersionCommand().processNodePaths(mockTerminal, nodePaths, environment));
+        assertThat(elasticsearchException.getMessage(), equalTo("aborted by user"));
+        assertThat(mockTerminal.getOutput(), allOf(
+            containsString("too old"),
+            containsString("data loss"),
+            containsString("You should not use this tool"),
+            containsString(Version.CURRENT.toString()),
+            containsString(nodeVersion.toString())));
+        expectThrows(IllegalStateException.class, () -> mockTerminal.readText(""));
+
+        final NodeMetaData nodeMetaData = NodeMetaData.FORMAT.loadLatestState(logger, xContentRegistry(), nodePaths);
+        assertThat(nodeMetaData.nodeId(), equalTo(nodeId));
+        assertThat(nodeMetaData.nodeVersion(), equalTo(nodeVersion));
+    }
+
+    public void testWarnsIfTooNew() throws Exception {
+        final String nodeId = randomAlphaOfLength(10);
+        final Version nodeVersion = NodeMetaDataTests.tooNewVersion();
+        NodeMetaData.FORMAT.writeAndCleanup(new NodeMetaData(nodeId, nodeVersion), nodePaths);
+        final MockTerminal mockTerminal = new MockTerminal();
+        mockTerminal.addTextInput("n\n");
```

Comment:
can do some randomization here as well

Reply:
Sure, done in e805ec3b3bb6b48bca043e6e223333ff6d911c8f.